R/lazyframe-frame.R

```diff
@@ -358,12 +769,1397 @@ lazyframe__slice <- function(offset, length = NULL) {
+#' observations based on matching observations, for example with `how =
+#' "inner"`).
+#'
+#' @inheritParams rlang::args_dots_empty
+#' @param other LazyFrame to join with.
+#' @param on Either a vector of column names or a list of expressions and/or
+#'   strings. Use `left_on` and `right_on` if the column names to match on are
+#'   different between the two DataFrames.
+#' @param how One of the following methods:
+#' * "inner": returns rows that have matching values in both tables
+#' * "left": returns all rows from the left table, and the matched rows from
+#'   the right table
+#' * "right": returns all rows from the right table, and the matched rows from
+#'   the left table
+#' * "full": returns all rows when there is a match in either left or right
+#'   table
+#' * "cross": returns the Cartesian product of rows from both tables
+#' * "semi": returns rows from the left table that have a match in the right
+#'   table.
+#' * "anti": returns rows from the left table that have no match in the right
+#'   table.
+#' @param left_on,right_on Same as `on` but only for the left or the right
+#'   DataFrame. They must have the same length.
+#' @param suffix Suffix to add to duplicated column names.
+#' @param validate Checks if join is of specified type:
+#' * `"m:m"` (default): many-to-many, doesn't perform any checks;
+#' * `"1:1"`: one-to-one, check if join keys are unique in both left and right
+#'   datasets;
+#' * `"1:m"`: one-to-many, check if join keys are unique in left dataset
+#' * `"m:1"`: many-to-one, check if join keys are unique in right dataset
+#'
+#' Note that this is currently not supported by the streaming engine.
+#'
+#' @param join_nulls Join on null values. By default null values will never
+#'   produce matches.
+#' @param allow_parallel Allow the physical plan to optionally evaluate the
+#'   computation of both DataFrames up to the join in parallel.
+#' @param force_parallel Force the physical plan to evaluate the computation of
+#'   both DataFrames up to the join in parallel.
+#' @param coalesce Coalescing behavior (merging of join columns).
+#' - `NULL`: join specific.
+#' - `TRUE`: Always coalesce join columns.
+#' - `FALSE`: Never coalesce join columns.
+#' Note that joining on any other expressions than `col` will turn off
+#' coalescing.
+#'
+#' @inherit as_polars_lf return
+#' @examples
+#' lf <- pl$LazyFrame(
+#'   foo = 1:3,
+#'   bar = c(6, 7, 8),
+#'   ham = c("a", "b", "c")
+#' )
+#' other_lf <- pl$LazyFrame(
+#'   apple = c("x", "y", "z"),
+#'   ham = c("a", "b", "d")
+#' )
+#' lf$join(other_lf, on = "ham")$collect()
+#'
+#' lf$join(other_lf, on = "ham", how = "full")$collect()
+#'
+#' lf$join(other_lf, on = "ham", how = "left", coalesce = TRUE)$collect()
+#'
+#' lf$join(other_lf, on = "ham", how = "semi")$collect()
+#'
+#' lf$join(other_lf, on = "ham", how = "anti")$collect()
+lazyframe__join <- function(
+    other,
+    on = NULL,
+    how = "inner",
```

Comment:
I may have asked this somewhere before, but is there any reason not to list them all here?

Reply:
Thanks, done in [`3f0af7f` (#49)](https://github.com/eitsupi/neo-r-polars/pull/49/commits/3f0af7f40ce5bc40b8e62c44ca69d739888cf573) (also for the two other comments).

> I may have asked this somewhere before, but is there any reason not to list them all here?

I think we decided to go with this before but it's just not in my habits. I don't really see the usecase but we already discussed this so no need to do it more ;)